### PR TITLE
chore: fix dependabot config

### PR DESCRIPTION
**Pull request by salmanm**
This PR contains config updates from https://github.com/fastify/fastify/pull/2672 as well as a new workflow that automatically merges dependabot PRs after CI goes green.

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: .github/workflows/dependabot-auto-merge.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      - synchronize
+      - unlabeled
+      - unlocked
+
+jobs:
+  worker:
+    runs-on: ubuntu-latest
+    steps:
+      - name: automerge
+        uses: ridedott/merge-me-action@master
```
Comment (jsumners) [inline .github/workflows/dependabot-auto-merge.yml:25]: Please use a specific version tag.

File: .github/workflows/dependabot-auto-merge.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# dependabot[bot]
+
+name: Dependabot Auto Merge
+
+on:
```
Comment (jsumners) [inline .github/workflows/dependabot-auto-merge.yml:5]: Where is the filter for the "dependabot" label?

Comment (L2jLiga) [reply .github/workflows/dependabot-auto-merge.yml:5]: I think it relies on author:

> GITHUB_LOGIN: dependabot[bot]

File: .github/workflows/dependabot-auto-merge.yml (hunk 1 above)
Comment (L2jLiga) [reply .github/workflows/dependabot-auto-merge.yml:25]: Yeah, I would also enable GitHub Actions autoupdate in dependabot

Comment (jsumners) [reply .github/workflows/dependabot-auto-merge.yml:25]: I wouldn't go that far. It doesn't honor beta stuff. Like right now, `setup-node@v1` is current but that bot will suggest updating to `setup-node@v2` which is in beta.

File: .github/workflows/dependabot-auto-merge.yml (hunk 2 above)
Comment (salmanm) [reply .github/workflows/dependabot-auto-merge.yml:5]: Yep, ridedott/merge-me-action checks for author. I tried adding a label check as well but it didn't work as expected for some reason. I believe the author check should be good enough.

**Review by L2jLiga — APPROVED** [review]
LGTM

File: .github/workflows/dependabot-auto-merge.yml (hunk 1 above)
Comment (L2jLiga) [reply .github/workflows/dependabot-auto-merge.yml]: I don't see anything bad in this beta, nvm, I'm ok to go with current solution

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by mcollina** [comment]
let's see if this works :).

**Pull request merged**